test/e2e/http2_disable_test.go

```diff
@@ -0,0 +1,245 @@
+		if err := client.Get(context.TODO(), defaultName, &ic); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", defaultName, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to get %q", err)
+	}
+
+	deployment := appsv1.Deployment{}
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		if err := client.Get(context.TODO(), controller.RouterDeploymentName(&ic), &deployment); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", controller.RouterDeploymentName(&ic), err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to get default ingresscontroller routerDeployment: %v", err)
+	}
+
+	ingressConfig := configv1.Ingress{}
+	if err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		name := types.NamespacedName{Name: "cluster"}
+		if err := client.Get(context.TODO(), name, &ingressConfig); err != nil {
+			t.Logf("Get %q failed: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to get ingress configuration: %v", err)
+	}
+
+	return &ic, &ingressConfig, &deployment, nil
+}
+
+func checkHTTP2IsEnabled(ic *operatorv1.IngressController, ingressConfig *configv1.Ingress, d *appsv1.Deployment) error {
+	icName := path.Join(ic.Namespace, ic.Name)
+	deploymentName := path.Join(d.Namespace, d.Name)
+	ingressConfigName := path.Join(ingressConfig.Namespace, ingressConfig.Name)
+
+	if ingress.HTTP2IsDisabledByAnnotation(ic.Annotations) {
+		return fmt.Errorf("expected ingress controller %q to have HTTP/2 enabled by default", icName)
+	}
+	if ingress.HTTP2IsDisabledByAnnotation(ingressConfig.Annotations) {
+		return fmt.Errorf("expected infrastructure config %q to have HTTP/2 enabled by default", ingressConfigName)
+	}
+	if len(d.Spec.Template.Spec.Containers) < 1 {
+		return fmt.Errorf("expected deployment %q to have at least one container", deploymentName)
+	}
+	if http2IsDisabledInEnv(d.Spec.Template.Spec.Containers[0].Env) {
+		return fmt.Errorf("expected deployment %q to have http/2 enabled by default", deploymentName)
+	}
+	return nil
+}
+
+func waitForRouterDeploymentHTTP2Disabled(client client.Client, timeout time.Duration, c *operatorv1.IngressController, expectedDisabledStatus bool) error {
+	return wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
+		deployment := &appsv1.Deployment{}
+		if err := client.Get(context.TODO(), controller.RouterDeploymentName(c), deployment); err != nil {
+			return false, nil
+		}
+		return http2IsDisabledInEnv(deployment.Spec.Template.Spec.Containers[0].Env) == expectedDisabledStatus, nil
+	})
+}
+
+func disableHTTP2ForIngressConfig(t *testing.T, client client.Client, timeout time.Duration, disableHTTP2 bool, c *configv1.Ingress) error {
+	name := types.NamespacedName{Namespace: c.Namespace, Name: c.Name}
```

Comment:
`t.Helper()`.

Reply:
These functions no longer call Fatal - they return a wrapped error. We carry t for logging through t.Logf().